src/driver/odoc_units_of.ml

```diff
@@ -283,10 +283,31 @@ let packages ~dirs ~extra_paths ~remap ~indices_style (pkgs : Packages.t list) :
                pkgs)
         in
         others @ List.concat std_units
-    | Normal | Voodoo | Automatic ->
+    | Normal _ | Voodoo | Automatic ->
         List.concat (pkg_index () :: src_index () :: std_units)
   in
-  if indices_style = Normal then
-    let gen_indices :> t = Landing_pages.package_list ~dirs ~remap pkgs in
-    gen_indices :: List.concat_map of_package pkgs
-  else List.concat_map of_package pkgs
+  match indices_style with
+  | Normal None ->
+      let gen_indices :> t = Landing_pages.package_list ~dirs ~remap pkgs in
+      gen_indices :: List.concat_map of_package pkgs
+  | Normal (Some index_mld) -> (
+      match Bos.OS.File.read index_mld with
+      | Error (`Msg msg) ->
+          Logs.err (fun m ->
+              m "Failed to read index_mld file '%a': %s" Fpath.pp index_mld msg);
+          []
```

Comment:
Return no pages, not even the pkgs pages ? I think the driver could abort with an exception instead.

Reply:
Hmm, I think what I'll do is move the `load` logic into `odoc_driver.ml`. The problem is that by the time we get here we've done a load of work and I didn't want to just exit. Something something sunk cost... :-)